pandas/_libs/tslibs/c_timestamp.pyx

```diff
@@ -119,7 +119,9 @@ cdef class _Timestamp(datetime):
             else:
                 return NotImplemented
 
-        self._assert_tzawareness_compat(other)
+        if op not in [Py_EQ, Py_NE]:
+            self._assert_tzawareness_compat(other)
+
```

Comment:
if we dont have tzawareness compat here, then eq should be False and ne should be True, i.e. dont just fall through to cmp_scalar here